src/app/client/src/app/modules/shared-feature/components/user-location/user-location.component.ts

```diff
@@ -259,21 +259,26 @@ export class UserLocationComponent implements OnInit {
   }
 
   clearInput(event, formControlName) {
-    if (event.target.value) {
-      this.userDetailsForm.get(formControlName).reset();
+    setTimeout(() => {
       switch (formControlName) {
         case 'state': {
-          this.stateDiv.query = '';
-          this.stateDiv.focus();
+          if (this.stateDiv.query && event.target.value) {
+            this.userDetailsForm.get(formControlName).reset();
+            this.stateDiv.query = '';
+            this.stateDiv.focus();
+          }
           break;
         }
         case 'district': {
-          this.districtDiv.query = '';
-          this.districtDiv.focus();
+          if (this.districtDiv.query && event.target.value) {
+            this.userDetailsForm.get(formControlName).reset();
+            this.districtDiv.query = '';
+            this.districtDiv.focus();
+          }
           break;
         }
       }
-    }
+    }, 500);
```

Comment:
Avoid timeout if we dont know the exact time. This may not work in all cases, especially in low configuration devices(desktops/laptops)